config/routes.rb

```diff
@@ -103,6 +103,7 @@
     member do
       get "deactivate", to: "users#deactivate"
       get "reactivate", to: "users#reactivate"
+      get "resend-invite", to: "users#resend_invite"
```

Comment:
It's just occurred to me that if we want to be properly RESTful, we may want to consider whether these should be get paths.
This is something I'm learning about at the moment so I would be interested in what @jamesrose or @RachaelBooth have to say about this.
I feel like deactivate and reactivate are changing the state of the model in question so they should probably be patch? (that would be outside the scope of this ticket anyway)
Does resend invite update anything on the user? Like a last invited field or anything? If an action has a side effect such as in this case sending an email ... ?
My instinct is a post would be most appropriate but I'm not confident in that!

Reply:
On deactivate and reactivate, you're right that we shouldn't expect a get request to be changing an object - but I think in this case those paths are actually opening an "are you sure" type page rather than being endpoints that directly do the de-/reactivation, which means that a get is expected even though it looks a little odd here.

Resend invite is a bit harder to obviously fit to an http verb, because it's doing an action (i.e. not just reading data) but not creating or updating the resource on our server that it is pointing at. I think it should be a post - the intended effect is to send an email which is not a read-only action and not one where repeating a request to the endpoint is without consequence (i.e. as long as we include emails to the user in things we're considering, this request is neither safe nor idempotent - and whatever we consider we don't want browsers treating it as something it can do without user input). 

Side node: this doesn't mean that gets can't have side effects - we could have counters or last read times or request logging or things like that being updated on a get and that would be fine, as long as there was nothing where we would care if it happened an extra time. (Also, a properly RESTful way might be to post to an email resource, but that is almost certainly not what we want to do.)

Reply:
ok great. very interesting. I haven't got a great handle on concept of idempotence. The resend method does update something on the user model, just to set something like initial_invitation_sent to true. As I understand it that is idempotent as however many times you click that button, it just gets set to true, so the same result.